resolwe_bio/processes/support_processors/merge_samples.py

```diff
@@ -0,0 +1,194 @@
+"""Merge FASTQs."""
+import glob
+import os
+from shutil import move
+
+from plumbum import TEE
+from resolwe.process import (
+    Cmd,
+    DataField,
+    FileField,
+    FileHtmlField,
+    ListField,
+    Process,
+    SchedulingClass,
+)
+
+
+class MergeFastqSingle(Process):
+    """Merge single-end FASTQs into one sample."""
+    slug = 'merge-fastq-single'
+    name = 'Merge FASTQ (single-end)'
+    process_type = 'data:reads:fastq:single'
+    version = '1.0.1'
+    category = 'Other'
+    scheduling_class = SchedulingClass.BATCH
+    entity = {
+        'type': 'sample',
+        'descriptor_schema': 'sample',
+        'always_create': True,
+    }
+    requirements = {
+        'expression-engine': 'jinja',
+        'executor': {
+            'docker': {
+                'image': 'resolwebio/common:1.3.1'
+            }
+        },
+    }
+    data_name = '{{ reads|map("sample_name")|join(", ")|default("?") }}'
+
+    class Input:
+        """Input fields to process MergeFastqSingle."""
+        reads = ListField(
+            DataField(data_type='reads:fastq:single:'),
+            label='Reads data objects',
+        )
+
+    class Output:
+        """Output fields to process MergeFastqSingle."""
+        fastq = ListField(FileField(), label='Reads file')
+        fastqc_url = ListField(
+            FileHtmlField(),
+            label='Quality control with FastQC',
+        )
+        fastqc_archive = ListField(FileField(), label='Download FastQC archive')
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+        fastqc = []
+        name = os.path.basename(inputs.reads[0].fastq[0].path).replace('.fastq.gz', '_merged')
+        merged_fastq = f'{name}.fastq.gz'
+        with open(merged_fastq, 'wb') as outfile:
+            for reads in inputs.reads:
+                for fastq in reads.fastq:
+                    with open(fastq.path, 'rb') as infile:
+                        for line in infile:
+                            outfile.write(line)
+
+        outputs.fastq = [merged_fastq]
+
+        print('Postprocessing FastQC...')
```

Comment:
FastQC operation is done twice in two processes within the same file. This could be generalised into function defined in this file. However, it is not yet possible to import this function from other location to be used in other processes too.